File: Leetcode/hashmap/1002_FindCommonCharacters/Solution.java
class Solution {
    public List<String> commonChars(String[] A) {
        List<String> res = new LinkedList<>();
        int[] count = new int[26];
        Arrays.fill(count, Integer.MAX_VALUE);
        for(String str: A){
            int[] temp = new int[26];
            str.chars().forEach(c-> ++temp[c-'a']);
            for(int i = 0; i<26; ++i){
                count[i] = Math.min(temp[i], count[i]);
            }
        }
        for(int i = 0; i<26; i++){
            while(count[i]>0){
                res.add(""+(char)('a'+i));
                count[i]--;
            }
        }
        return res;
    }
}

File: Leetcode/hashmap/1002_FindCommonCharacters/Solution.py
class Solution(object):
    def commonChars(self, A):
        """
        :type A: List[str]
        :rtype: List[str]
        """
        res = collections.Counter(A[0])
        for a in A:
            res &= collections.Counter(a)
        return list(res.elements())
